File: src/main/java/cn/solarmoon/idyllic_food_diary/element/matter/cookware/steamer/SteamerInventory.java
package cn.solarmoon.idyllic_food_diary.element.matter.cookware.steamer;

import cn.solarmoon.solarmoon_core.api.tile.inventory.TileInventory;
import net.minecraft.core.NonNullList;
import net.minecraft.world.item.ItemStack;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.stream.Collectors;

public class SteamerInventory extends TileInventory {

    public SteamerInventory(int size, SteamerBlockEntity steamer) {
        super(size, 1, steamer);
    }

    /**
     * @return 删除所有非空元素的新stacks
     */
    public NonNullList<ItemStack> getStacks() {
        return stacks;
    }

}

File: src/main/kotlin/cn/solarmoon/idyllic_food_diary/element/matter/cookware/steamer/SteamerInventory.kt
package cn.solarmoon.idyllic_food_diary.element.matter.cookware.steamer

import cn.solarmoon.idyllic_food_diary.registry.common.IFDRecipes
import net.minecraft.world.item.ItemStack
import net.neoforged.neoforge.items.ItemStackHandler

class SteamerInventory(size: Int): ItemStackHandler(size) {

    private var steamer: SteamerBlockEntity? = null

    override fun isItemValid(slot: Int, stack: ItemStack): Boolean {
        val level = steamer?.level ?: return false
        val recipes = level.recipeManager.getAllRecipesFor(IFDRecipes.STEAMING.type.get()).map { it.value }
        return recipes.any { it.input.test(stack) }
    }

    override fun getSlotLimit(slot: Int): Int {
        return 1
    }

    fun applyTile(steamer: SteamerBlockEntity) = apply {
        this.steamer = steamer
    }

    override fun onContentsChanged(slot: Int) {
        super.onContentsChanged(slot)
        steamer?.invalidateCapabilities()
        steamer?.setChanged()
    }

}